packages/nuxt/src/core/nuxt.ts

```diff
@@ -621,4 +628,15 @@ export async function loadNuxt (opts: LoadNuxtOptions): Promise<Nuxt> {
   return nuxt
 }
 
+async function checkDependencyVersion (name: string, nuxtVersion: string): Promise<void> {
+  const path = await resolvePath(name).catch(() => null)
+
+  if (!path) { return }
+  const { version } = await readPackageJSON(path)
+
+  if (version !== nuxtVersion) {
+    console.warn(`Version mismatch for ${name} and nuxt: expected ${nuxtVersion} (nuxt) but got ${version}. This might lead to unexpected behavior. Check your package.json or refresh your lockfile.`)
```

Comment:
Might be worth adding a few lines in the docs, and linking to it so we can update with workarounds or more explanation later on.